File: src/Days 1 to 10/Day 10/python/main.py
def anagramCheck(first, second):
    if len(first) != len(second):
        return False
    
    first_dict, second_dict = {}, {}
    for element in first:
        first_dict[element] = first_dict.get(element, 0) + 1
    for element in second:
        second_dict[element] = second_dict.get(element, 0) + 1
    
    return first_dict == second_dict

s = "anagram"
t = "nagaram"
result = anagramCheck(s, t)
if result:
    print("yes")
else:
    print("no")

s = "anagra"
t = "nagaram"
result = anagramCheck(s, t)
if result:
    print("yes")
else:
    print("no")

File: src/Days 1 to 10/Day 10/kotlin/main.kt
fun anagramCheck(first: String, second: String): Boolean {
    if (first.length != second.length) return false
    
    val firstDict = mutableMapOf<Char, Int>()
    val secondDict = mutableMapOf<Char, Int>()

    for (char in first) {
        firstDict[char] = firstDict.getOrDefault(char, 0) + 1
    }

    for (char in second) {
        secondDict[char] = secondDict.getOrDefault(char, 0) + 1
    }

    return firstDict == secondDict
}

fun main() {
    var s = "anagram"
    var t = "nagaram"
    var result = anagramCheck(s, t)
    if (result) {
        println("yes")
    } else {
        println("no")
    }

    s = "anagra"
    t = "nagaram"
    result = anagramCheck(s, t)
    if (result) {
        println("yes")
    } else {
        println("no")
    }
}